### ISSUE TITLE
getProjectWideFailFast

### ISSUE THREAD

**Issue by jesperancinha** [kotest/kotest-extensions-spring#22]
Hi there,

Ever since Dependabot update `kotest-extensions-spring` in my project with version 1.1.0, it seems like I can't build my project anymore with this bizarre Exception:

```java
java.lang.NoSuchMethodError: 'boolean io.kotest.core.config.ProjectConfiguration.getProjectWideFailFast()'
	at io.kotest.engine.ConcurrentTestSuiteScheduler$schedule$8$1$2.invokeSuspend(ConcurrentTestSuiteScheduler.kt:65)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
	at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
	at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
	at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source)
	at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
	at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source)
	at io.kotest.common.RunBlockingKt.runBlocking(runBlocking.kt:3)
	at io.kotest.engine.launcher.MainKt.main(main.kt:34)

'boolean io.kotest.core.config.ProjectConfiguration.getProjectWideFailFast()
```

It seems like, no matter what I do to update, follow instructions (https://kotest.io/docs/extensions/spring.html), Java versions (11 and 17, don't get me started on Java 8 please), the tests seem to just keep failing. Could you please let me know what am I missing or if this is a bug on this library? Thanks in advance!

**Comment by sksamuel** [comment]
What version of kotest?

**Comment by jesperancinha** [comment]
I'm using `5.1.0`

> What version of kotest?

**Comment by sksamuel** [comment]
It looks like a 4.x version is sneaking in somewhere.

**Comment by jesperancinha** [comment]
I can't see any 4.x version my dependency tree. Maybe it would help if you tell me what problems were there with version 4.x? I did see  that this should work with version 5, but I'm using `5.1.0`. Maybe that has an influence? Anyways here are my dependencies as of now:

```xml
    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-jpa</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlin</groupId>
            <artifactId>kotlin-stdlib</artifactId>
            <version>1.6.20-RC</version>
        </dependency>
        <dependency>
            <groupId>org.jetbrains.kotlin</groupId>
            <artifactId>kotlin-reflect</artifactId>
            <version>1.6.20-RC</version>
        </dependency>
        <dependency>
            <groupId>io.kotest.extensions</groupId>
            <artifactId>kotest-extensions-spring</artifactId>
            <version>1.1.0</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>io.kotest</groupId>
            <artifactId>kotest-runner-junit5-jvm</artifactId>
            <version>5.1.0</version>
            <scope>test</scope>
        </dependency>
        <!-- https://mvnrepository.com/artifact/io.kotest/kotest-framework-engine-jvm -->
        <dependency>
            <groupId>io.kotest</groupId>
            <artifactId>kotest-framework-engine-jvm</artifactId>
            <version>5.1.0</version>
            <scope>test</scope>
        </dependency>

        <dependency>
            <groupId>io.mockk</groupId>
            <artifactId>mockk</artifactId>
            <version>1.12.3</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-test</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-test-autoconfigure</artifactId>
            <scope>test</scope>
        </dependency>
    </dependencies>
```

Spring boot is `2.6.4`. It could just be something from the latest versions. That's why I'm asking about 4.x. Maybe there could be a clue about solving this

**Comment by jesperancinha** [comment]
Do you have an idea about this yet? Otherwise, maybe I can help and open a branch myself about this to find the issue and possibly fix it?

**Comment by sksamuel** [comment]
I'm 99% sure that an earlier version of kotest is in your build, since that projectWideFailFast was only added in 5.1.0 and a NoSuchMethodError usually means that the compile time and runtime versions are different.

I doubt spring itself is doing this as I don't think it brings in kotest. Do you have kotest defined elsewhere in the project ?

**Comment by jesperancinha** [comment]
@sksamuel , I finally found a solution for this issue. I frankly do not understand what happened and why is this a solution. Indeed I had this 5.0.0 version in my effective POM:

```xml
<dependency>
    <groupId>io.kotest</groupId>
    <artifactId>kotest-framework-api-jvm</artifactId>
    <version>5.0.0</version>
</dependency>
```

In my normal POM,  I'm not even declaring this dependency. And Intellij doesn't even find it in the dependency overview even after a full refresh. I only added this to my declared POM:

```xml
    <dependency>
        <groupId>io.kotest</groupId>
        <artifactId>kotest-framework-api-jvm</artifactId>
        <version>5.2.1</version>
    </dependency>
```

And now it works. How 5.0.0 ends up creeping in there anyways I do not know, and it's quite strange since I'm not using that many dependencies. 4.X is definitely not in my project setup. This API dependency was really where this problem began.
